src/aeonlib/models.py

```diff
@@ -40,66 +40,134 @@ class SiderealTarget(BaseModel):
-    """The name of this Target"""
-    type: Literal["ORBITAL_ELEMENTS", "SATELLITE"]
-    """The type of this Target TODO: Where does HOUR_ANGLE make sense?"""
-    scheme: Literal[
-        "ASA_MAJOR_PLANET",
-        "ASA_MINOR_PLANET",
-        "ASA_COMET",
-        "JPL_MAJOR_PLANET",
-        "JPL_MINOR_PLANET",
-        "MPC_MINOR_PLANET",
-        "MPC_COMET",
-    ]
-    """The Target scheme to use"""
+    start: Time | None = None
+    end: Time
+
+
+class _NonSiderealTarget(BaseModel):
+    """Base class for non-sidereal targets, should not be used directly"""
+
+    model_config = ConfigDict(validate_assignment=True)
+    type: Literal["ORBITAL_ELEMENTS"] = "ORBITAL_ELEMENTS"
+    name: Annotated[str, StringConstraints(max_length=50)]
     epochofel: TimeMJD
     """The epoch of the orbital elements (MJD)"""
     orbinc: Angle
     """Orbital inclination (angle in degrees)"""
     longascnode: Angle
     """Longitude of ascending node (angle in degrees)"""
-    argofperih: Angle
-    """Argument of perihelion (angle in degrees)"""
     eccentricity: NonNegativeFloat
     """Eccentricity of the orbit"""
-    meandist: Annotated[float, NonNegativeFloat] | None = None
-    """Semi-major axis (AU)"""  # Not Comet
-    meananom: Angle | None = None
-    """Mean anomaly (angle in degrees)"""
-    perihdist: Annotated[float, NonNegativeFloat] | None = None
-    """Perihelion distance (AU)"""  # Comet Only
-    epochofperih: TimeMJD | None = None
-    """Epoch of perihelion (MJD)"""  # Comet Only
-    dailymot: float | None = None
-    """Daily motion (angle in degrees)"""  # Major Planet Only
-    altitude: Angle | None = None
-    """Altitude of this Target in decimal degrees"""  # Satellite Only
-    azimuth: Angle | None = None
-    """Azimuth of this Target in decimal degrees east of North"""  # Satellite Only
-    diff_altitude_rate: float | None = None
-    """Differential altitude rate (arcsec/s)"""  # Satellite Only
-    diff_azimuth_rate: float | None = None
-    """Differential azimuth rate (arcsec/s)"""  # Satellite Only
-    diff_epoch: float | None = None
-    """Reference time for non-sidereal motion (MJD)"""  # Satellite Only
-    diff_altitude_acceleration: float | None = None
-    """Differential altitude acceleration (arcsec/s^2)"""  # Satellite Only
-    diff_azimuth_acceleration: float | None = None
-    """Differential azimuth acceleration (arcsec/s^2)"""  # Satellite Only
-    meanlong: Angle | None = None
-    """Mean longitude (angle in degrees)"""  # Major Planet Only
-    longofperih: Angle | None = None
-    """Longitude of perihelion (angle in degrees)"""  # Major Planet Only
     extra_params: dict[Any, Any] = {}
 
 
-class Window(BaseModel):
-    """A general time window"""
+class AsaMajorPlanetTarget(_NonSiderealTarget):
+    scheme: Literal["ASA_MAJOR_PLANET"] = "ASA_MAJOR_PLANET"
+    longofperih: Angle
+    """Longitude of perihelion (angle in degrees)"""
```

Comment:
This might just be me not understanding how this works, but does this still need to be "angle in degrees"?

Reply:
I'm assuming you are referring to the comment, yes that's outdated as you should be able to supply the Angle however you want.

Reply:
I removed all the (angle in degrees) comments for Angle types.